### PR TITLE
Transfer replacement

### PR DESCRIPTION
**Pull request by nventuro**
This PR adds a replacement for Solidity's `transfer`, which imposes a fixed gas limit of 2300. While useful to prevent reentrancy issues, this has proved to be an issue due to gas repricings, like the one [EIP1884](https://eips.ethereum.org/EIPS/eip-1884) will introduce on the Instanbul hardfork.

Given that the code to imitate `transfer` is tricky and cryptic, having a helper method to do it is a good idea.

### CODE REVIEW

**Comment by nventuro** [comment]
This should go into the 2.4 release, but I started the branch from `master` instead of `release-2.4`. Should I rebase these commits on top of 2.4, or do we want to simply cherry-pick the merge commit there afterwards?

**Comment by frangio** [comment]
Cherry-picking aftewards is the "protocol".

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 ### New features:
  * `Address.toPayable`: added a helper to convert between address types without having to resort to low-level casting. ([#1773](https://github.com/OpenZeppelin/openzeppelin-solidity/pull/1773))
  * Facilities to make metatransaction-enabled contracts through the Gas Station Network. ([#1844](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/1844))
+ * `Address.sendEther`: added a replacement to Solidity's `transfer`, removing the fixed gas stipend. ([#1961](https://github.com/OpenZeppelin/openzeppelin-solidity/pull/1961))
```
Comment (jmendiola222) [inline CHANGELOG.md]: Isn't the "Ether" naming too network coupled? I mean this aiming to be standard solidity contracts that should work with any compible EVM. (RSK or Cardano just to name a few)

Comment (nventuro) [reply CHANGELOG.md]: Hm, this is a good point. What would you suggest? `sendFunds`?

File: test/utils/Address.test.js [hunk 2]
```diff
@@ -35,4 +36,70 @@ contract('Address', function ([_, other]) {
...
+
+      it('reverts when sending non-zero amounts', async function () {
+        await expectRevert(this.mock.sendEther(other, 1), 'Address: not enough ether to send');
+      });
+    });
+
+    context('when sender contract has funds', function () {
+      const funds = ether('1');
+      beforeEach(async function () {
+        await send.ether(other, this.mock.address, funds);
```
Comment (frangio) [inline test/utils/Address.test.js]: Shouldn't this be affecting the balance tracker above?

File: contracts/utils/Address.sol [hunk 3]
```diff
@@ -40,4 +40,28 @@ library Address {
...
+     * `transfer`. {sendEther} removes this limitation.
+     *
+     * https://diligence.consensys.net/posts/2019/09/stop-using-soliditys-transfer-now/[Learn more].
+     *
+     * IMPORTANT: because control is transferred to `recipient`, care must be
+     * taken to not create reentrancy vulnerabilities. Consider using
+     * {ReentrancyGuard} or the
+     * https://solidity.readthedocs.io/en/v0.5.11/security-considerations.html#use-the-checks-effects-interactions-pattern[checks-effects-interactions pattern].
+     */
+    function sendEther(address payable recipient, uint256 amount) internal {
```
Comment (frangio) [inline contracts/utils/Address.sol]: Let's rename to `sendValue` to address the issue mentioned above.

File: test/utils/Address.test.js (hunk 2 above)
Comment (nventuro) [reply test/utils/Address.test.js]: The tracker is for `recipient`, this line funds the contract from `other`, so it doesn't affect it.

File: CHANGELOG.md [hunk 4]
```diff
@@ -10,6 +10,7 @@
 ### New features:
  * `Address.toPayable`: added a helper to convert between address types without having to resort to low-level casting. ([#1773](https://github.com/OpenZeppelin/openzeppelin-solidity/pull/1773))
  * Facilities to make metatransaction-enabled contracts through the Gas Station Network. ([#1844](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/1844))
+ * `Address.sendValue`: added a replacement to Solidity's `transfer`, removing the fixed gas stipend. ([#1961](https://github.com/OpenZeppelin/openzeppelin-solidity/pull/1961))
```
Comment (igorline) [inline CHANGELOG.md]: @nventuro should it be 1962?

Comment (frangio) [reply CHANGELOG.md]: Yes, good catch. I'll fix it on `master`.

**Pull request merged**